client/post-editor/editor-ground-control/save-status.jsx

```diff
@@ -0,0 +1,56 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+import { get, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import postUtils from 'lib/posts/utils';
+
+// First, validate that the post had an ID, presumably this
+// allows us to determine whether or not it is stored as a draft.
+// More semantic way to express this??
+const isPostNotYetPublished = post => get( post, 'ID' ) && ! postUtils.isPublished( post );
```

Comment:
Can't we avoid this function altogether by not `render`ing `SaveStatus` in the parent component when `get( post, 'ID' )` is falsey?

Reply:
Though that would be true, one of the main motives for moving this into a new component was to defer the heavy logic to this file in an attempt to keep `editor-ground-control/index.jsx` from becoming more complex.
Also, given that this component can re-render often (consider the flow of autosaving often, going from 'saving...' to 'saved' repeatedly) I think we'd actual see benefit in react keeping a `null` rendered component ref and updating it rather than creating new components each time the `conditional && <Component>` condition is true.